src/packaging/metadata.rs

```diff
@@ -58,42 +58,46 @@ pub fn contains_prefix_binary(file_path: &Path, prefix: &Path) -> Result<bool, P
 
 /// This function requires we know the file content we are matching against is
 /// UTF-8 In case the source is non utf-8 it will fail with a read error
+///
+/// Returns all variations of the expanded prefix found in the file.
 pub fn contains_prefix_text(
     file_path: &Path,
     prefix: &Path,
     target_platform: &Platform,
-) -> Result<Option<String>, PackagingError> {
-    // Open the file
-    let file = File::open(file_path)?;
-
-    // mmap the file
-    let mmap = unsafe { memmap2::Mmap::map(&file)? };
-
-    // Check if the content contains the prefix with memchr
-    let prefix_string = prefix.to_string_lossy().to_string();
-    if memchr::memmem::find_iter(mmap.as_ref(), &prefix_string)
-        .next()
-        .is_some()
-    {
-        return Ok(Some(prefix_string));
-    }
+) -> Result<Vec<String>, PackagingError> {
+    let mut found_variations = Vec::new();
 
-    if target_platform.is_windows() {
-        use crate::utils::to_forward_slash_lossy;
-        // absolute and unc paths will break but it,
-        // will break either way as C:/ can't be converted
-        // to something meaningful in unix either way
-        let forward_slash: Cow<'_, str> = to_forward_slash_lossy(prefix);
+    {
+        // Open the file in a new scope to ensure the memory map and file handle are dropped
```

Comment:
I don't get that, right after the end of the scope, the scope of the function ends as well.